search/api.py

```diff
@@ -73,6 +81,59 @@ def gen_course_id(course_id):
     return "co_{}".format(safe_id)
 
 
+def gen_bootcamp_id(bootcamp_id):
+    """
+    Generates the Elasticsearch document id for a bootcamp
+
+    Args:
+        bootcamp_id (str): The course_id of a Bootcamp object
+
+    Returns:
+        str: The Elasticsearch document id for this object
+    """
+    safe_id = urlsafe_b64encode(bootcamp_id.encode("utf-8")).decode("utf-8").rstrip("=")
+    return "b_{}".format(safe_id)
+
+
+def gen_program_id(program_obj):
+    """
+    Generates the Elasticsearch document id for a program
+
+    Args:
+        program_obj (Program): The Program object
+
+    Returns:
+        str: The Elasticsearch document id for this object
+    """
+    safe_id = (
+        urlsafe_b64encode(program_obj.title.encode("utf-8"))
+        .decode("utf-8")
+        .rstrip("=")
+        .replace(" ", "_")
+    )
+    return "prog_{}".format(safe_id)
```

Comment:
What should be the id for a Program? Will titles for Programs be unique (in which case, we can add a database-enforced unique constraint to the model)?

Reply:
Another concern besides uniqueness is - what if the program title needs to be modified?  Ideally the id would be something that would never change.

Reply:
Indeed. What else could we use?

Reply:
As discussed in-person, we'll be using the autogenerated object ID.